### PR TITLE
migration to unified syntax of blackbox tests for price aggregator

### PR DESCRIPTION
**Pull request by mihaicalinluca**

### CODE REVIEW

File: contracts/core/price-aggregator/tests/price_aggregator_blackbox_test.rs [hunk 1]
```diff
@@ -148,27 +158,31 @@ impl PriceAggregatorTestState {
...
     fn vote_slash_member(&mut self, from: &AddressValue, member_to_slash: Address) {
-        self.world.sc_call(
-            ScCallStep::new().from(from).call(
-                self.price_aggregator_contract
-                    .vote_slash_member(member_to_slash),
-            ),
-        );
+        self.world
+            .tx()
+            .from(&from.to_address())
```
Comment (andrei-marinica) [inline contracts/core/price-aggregator/tests/price_aggregator_blackbox_test.rs:181]: Should work directly as .from(from)

If it doesn't, then we need some more conversions.

File: contracts/core/price-aggregator/tests/price_aggregator_blackbox_test.rs [hunk 2]
```diff
@@ -62,7 +63,7 @@ impl PriceAggregatorTestState {
         }
         world.set_state_step(set_state_step);
 
-        let price_aggregator_contract = PriceAggregatorContract::new(PRICE_AGGREGATOR_ADDRESS_EXPR);
+        // let price_aggregator_contract = PriceAggregatorContract::new(PRICE_AGGREGATOR_ADDRESS_EXPR);
```
Comment (andrei-marinica) [inline contracts/core/price-aggregator/tests/price_aggregator_blackbox_test.rs:66]: Clean up

File: contracts/core/price-aggregator/tests/price_aggregator_blackbox_test.rs [hunk 3]
```diff
@@ -361,13 +375,14 @@ fn test_price_aggregator_slashing() {
...
-        ScCallStep::new().from(&state.oracles[0]).call(
-            state
-                .price_aggregator_contract
-                .slash_member(state.oracles[1].to_address()),
-        ),
-    );
+    state
+        .world
+        .tx()
+        .from(&state.oracles[0].to_address())
```
Comment (andrei-marinica) [inline contracts/core/price-aggregator/tests/price_aggregator_blackbox_test.rs:381]: Same, should work without .to_address()

**Review by andrei-marinica — CHANGES_REQUESTED** [review]

**Review by BiancaIalangi — DISMISSED** [review]

File: contracts/core/price-aggregator/tests/price_aggregator_proxy.rs [hunk 4]
```diff
@@ -0,0 +1,366 @@
...
+        member_to_slash: Arg0,
+    ) -> TxProxyCall<Env, From, To, Gas, ()> {
+        self.wrapped_tx
+            .raw_call("slashMember")
+            .argument(&member_to_slash)
+            .original_result()
+    }
+}
+
+#[derive(TopEncode, TopDecode)]
```
Comment (andrei-marinica) [inline contracts/core/price-aggregator/tests/price_aggregator_proxy.rs:343]: I think the original version had more derives. Extract `#[type_abi]`

File: contracts/core/price-aggregator/tests/price_aggregator_proxy.rs [hunk 5]
```diff
@@ -0,0 +1,366 @@
...
+    pub round_id: u32,
+    pub from: ManagedBuffer<Api>,
+    pub to: ManagedBuffer<Api>,
+    pub timestamp: u64,
+    pub price: BigUint<Api>,
+    pub decimals: u8,
+}
+
+#[derive(TopEncode, TopDecode)]
+pub struct NewRoundEvent<Api>
```
Comment (andrei-marinica) [inline contracts/core/price-aggregator/tests/price_aggregator_proxy.rs:357]: I think the original version had more derives. Extract `#[type_abi]`

**Review by andrei-marinica — CHANGES_REQUESTED** [review]

**Review by andrei-marinica — APPROVED** [review]

**Pull request merged**